ontolib-core/src/main/java/com/github/phenomics/ontolib/ontology/data/ImmutableOntology.java

```diff
@@ -188,6 +185,25 @@ public TermId getRootTermId() {
     return obsoleteTermIds;
   }
 
+  @Override
+  public boolean existsPath(final TermId sourceID, TermId destID){
+    // special case -- a term cannot have a path to itself in an ontology (DAG)
+    if (sourceID.equals(destID)) return false;
```

Comment:
Please wrap and indent consistently. I can send around a style file for Eclipse.